Fix background colour for Viewer pane

timtmok opened the pull request:
Address #3759 

Sets the background colour on the viewer. When a webview's content doesn't set the background colour on the `body` or `html` tags, the background becomes the theme's background colour. This didn't work well when the content assumes a light theme.

This forces the background to be white for these cases. If the content is a dark theme, it is assumed that it sets the background colour on the webview and will override the colour in the view. For other content that does not set a background, it becomes transparent and takes on the default white. Browsers commonly do this.

Preview in Viewer
![image](https://github.com/user-attachments/assets/afa92cd9-46cb-4d75-8222-329fd63736b0)

![image](https://github.com/user-attachments/assets/3da1572e-5d3a-4fa1-b1ef-4bfb7091ceb7)

Preview in editor
![image](https://github.com/user-attachments/assets/2ba6f51c-a237-43ec-92f7-160069c70d14)


### Release Notes

<!--
  Optionally, replace `N/A` with text to be included in the next release notes.
  The `N/A` bullets are ignored. If you refer to one or more Positron issues,
  these issues are used to collect information about the feature or bugfix, such
  as the relevant language pack as determined by Github labels of type `lang: `.
  The note will automatically be tagged with the language.

  These notes are typically filled by the Positron team. If you are an external
  contributor, you may ignore this section.
-->

#### New Features

- N/A

#### Bug Fixes

- Fix Viewer style in dark mode with content that does not set a background style #3759 


### QA Notes
Check the issue for code examples to reproduce the issue.
<!--
  Add additional information for QA on how to validate the change,
  paying special attention to the level of risk, adjacent areas that
  could be affected by the change, and any important contextual
  information not present in the linked issues.
-->

timtmok committed:
Fix background colour for Viewer pane

timtmok committed:
Fix preview in editor

jennybc reviewed (dismissed):
LGTM and I can definitely see a big improvement, using some of the examples in the original issue.

The one thing that I wonder about is whether we want to paint the Viewer white immediately and unconditionally like this.

Here's what I see when I just fire up Positron.

<img width="1240" alt="Screenshot 2025-05-15 at 10 37 20 AM" src="https://github.com/user-attachments/assets/512fba1f-eb4f-45ad-bb3d-1982c7c50d17" />

That giant white rectangle really stands out when I'm using dark mode. I'm not sure what to do about this, but wanted to discuss.

timtmok commented:
The webview is overlaid on top of the Viewer pane. So it isn't a child in the DOM because it's positioned with absolute values. I'm not quite sure how to determine if something is rendered but I'll take a look. It might be possible to add some more class names to the div when there is something to look at and then show a white background.

timtmok committed:
Fix blank Viewer background colour

timtmok committed:
Remove unnecessary css

timtmok commented:
@jennybc I changed the styling and added a class on the webview. This allows the Viewer to use the theme's background colour until a webview is shown.

![image](https://github.com/user-attachments/assets/7f143c39-3a82-4fae-a3b6-e587d7134eb7)

jennybc reviewed (approved):
I haven't pulled this branch down again to experience it, but the screenshot looks great! I think this is a worthwhile tweak.
